docs/pages/technical-specs/expo-updates-0.md

```diff
@@ -120,20 +113,50 @@ type Asset = {
     * `hash`: SHA-256 hash of the file to guarantee integrity.
     * `key`: Key used to reference this asset from the update's application code. This key, for example, may be generated by a separate build step that processes the application code, such as a bundler.
     * `contentType`: The MIME type of the file as defined by [RFC 2045](https://tools.ietf.org/html/rfc2045). e.g. `application/javascript`, `image/jpeg`.
-    * `fileExtension`: The suggested extension to use when a file is saved on a client. Some platforms, such as iOS, require certain file types to be saved with an extension. The extension MUST be prefixed with a `.`. e.g. **.jpeg**. In some cases, such as the launchAsset, this field will be ignored in favor of a locally determined extension. If the field is omitted and there is no locally stipulated extension, the asset will be saved without an extension, e.g. `./filename` with no `.` at the end. 
+    * `fileExtension`: The suggested extension to use when a file is saved on a client. Some platforms, such as iOS, require certain file types to be saved with an extension. The extension MUST be prefixed with a `.`. e.g. **.jpeg**. In some cases, such as the launchAsset, this field will be ignored in favor of a locally determined extension. If the field is omitted and there is no locally stipulated extension, the asset will be saved without an extension, e.g. `./filename` with no `.` at the end.
     A conforming client SHOULD prefix a file extension with a `.` if a file extension is not empty and missing the `.` prefix.
     * `url`: Location at which the file may be fetched.
   * `metadata`: The metadata associated with an update. It is a string-valued dictionary. The server MAY send back anything it wishes to be used for filtering the updates. The metadata MUST pass the filter defined in the accompanying `expo-manifest-filters` header.
   * `extra`: For storage of optional "extra" information such as third-party configuration. For example, if the update is hosted on Expo Application Services (EAS), the EAS project ID may be included:
-  ```typescript
-  extra: {
-    eas: {
-      projectId: '00000000-0000-0000-0000-000000000000'
+  ```json
+  "extra": {
+    "eas": {
+      "projectId": "00000000-0000-0000-0000-000000000000"
     }
-    ...
   }
   ```
 
+### Multipart Manifest Response
+
+A manifest response of this format is defined by the `multipart/mixed` MIME type as defined by [RFC 2046](https://tools.ietf.org/html/rfc2046#section-5.1). Each part is defined as follows:
+1. REQUIRED `"manifest"` part:
+    - MUST have header `content-disposition: inline; name="manifest"` to identify this part.
+    - The [manifest headers](#manifest-response-headers) MUST be sent in the part headers.
+    - The [manifest body](#manifest-response-body) MUST be sent in the part body.
+2. OPTIONAL `"extensions"` part:
```

Comment:
So the case you are thinking about is the client asks for multipart, but unsigned? I don't understand this use case🤔

> conformant servers to only implement the mixed response type
as its written, this doesn't seem allowed by the spec?

Reply:
You bring up a good point which is the question: is it considered conformant if a client and/or server implementation don't support multipart or don't support normal JSON? Is sending a 406 in either of these cases conformant?

The next logical question is whether we want our implementations to be conformant (we should IMO). Our client implementation (the expo client) will support both and `accept` both. We were thinking of making our server only support multipart since we need it for asset HMACs and all our assets need HMACs (we can't serve a normal manifest). But then comes a question: what if a custom client implementation wants to talk to the EAS server implementation? How do we specify in the spec that this is possible even if we can't serve normal JSON?

Reply:
We probably will change the spec to say the server can respond with a 406 or the content type it supports.

An extensionless multipart response doesn't cause any problems and doesn't make the client code more difficult to write since we need to handle the case where "extensions" could be missing (don't want to crash the app).

Reply:
> We probably will change the spec to say the server can respond with a 406 or the content type it supports.

This spec change is probably the most appropriate time and place to make this decision.

- If we say that a conformant server can respond with a 406, that means that it is ok for a conformant client and conformant server never to be able to communicate and get updates which IMO seems to be a pretty big issue.
- If we say that a conformant server can respond with a content type it supports even if it wasn't in the request accept header, then that means it is ok for a conformant client to receive an update it doesn't know how to comprehend.
- Another option is that we could say that a conformant client must support both JSON and multipart and always `accept` both.

Reply:
I don't see the 406 as a big issue. You'd find this out in development when picking what server to use.

Clients already need to handle responses they don't understand e.g. the 4xx HTML pages from NGINX.

Reply:
Yeah, that's fair. I'll add a blurb along the lines of "A conformant server may support either or both, and a client may support either or both. When a server is unable to respond to a client, a 406 should be returned".